lib_xcore_math/src/bfp/bfp_s32.c

```diff
@@ -335,8 +335,8 @@ float_s32_t bfp_s32_mean(
     right_shift_t shr = MAX(0, 32 - HR_S64(mean));
 
     if(shr > 0)
-        mean += 1 << (shr-1);
-    
+        mean += (uint32_t)(1 << (shr-1));
```

Comment:
can you add a comment to explain why you have to cast a uint32_t value into uint64_t?

Reply:
I did a slight update on this line. I don't really know how to comment it, as it's a standard procedure for rounding